src/ucs/stats/stats.c

```diff
@@ -460,6 +476,184 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
     }
 }
 
+static uint32_t
+ucs_stats_clsid_to_agrgt_sum_base_offset_get(ucs_stats_node_t *node,
+                                             size_t num_counters_in_class)
+{
+    uint32_t offset                      = 0;
+    ucs_stats_filter_node_t *filter_node = node->filter_node;
+    const char *class_name               = node->cls->name;
+    unsigned clsid                       = node->cls->class_id;
+    unsigned *class_id                   = &node->cls->class_id;
+    ucs_stats_node_t *temp_node;
+    uint32_t counter_index;
+    uint32_t cnt_agrgt_sum_index;
+
+    pthread_mutex_lock(&ucs_stats_context.lock);
+
+    if (ucs_likely(clsid != 0) && (ucs_stats_context.aggrgt_sum_db)) {
```

Comment:
please compare for NULL explicitly, like `aggrgt_sum_db != NULL`

Reply:
1. unhandled comment above
2. why need to check (ucs_stats_context.aggrgt_sum_db) ? could it be NULL if clsid!=0?
3. could be like this, to simplify the "good flow":
````
if (ucs_likely(...)) {
   offset = ucs_stats_context.aggrgt_sum_db[clsid].offset
   goto out
}
```